lib/Basics/FileUtils.cpp

```diff
@@ -804,4 +804,100 @@ std::string slurpProgram(std::string const& program) {
   return output;
 }
 
+}  // namespace
+
+std::string slurpProgram(std::string const& program) {
+  std::vector<std::string> moreArgs{std::string("version")};
+  return slurpProgramInternal(program, moreArgs);
+}
+
+#ifdef ARANGODB_HAVE_GETPWUID
+std::optional<uid_t> findUser(std::string const& nameOrId) noexcept {
+  // We avoid getpwuid and getpwnam because they pose problems when
+  // we build static binaries with glibc (because of /etc/nsswitch.conf).
+  // However, we know that `id` exists for basically all Linux variants
+  // and for Mac.
+  try {
+    std::vector<std::string> args{"-u", nameOrId};
+    std::string output = slurpProgramInternal("/usr/bin/id", args);
+    StringUtils::trimInPlace(output);
+    bool valid = false;
+    uid_t uidNumber = NumberUtils::atoi_positive<int>(
+        output.data(), output.data() + output.size(), valid);
+    if (valid) {
+      return {uidNumber};
+    }
+  } catch (std::exception const& exc) {
```

Comment:
I have seen MSVC complaining about unused exception variables before. Better not use it.
```suggestion
  } catch (std::exception const&) {
```